tests/test_threads.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+import os
+from typing import Any, Dict, List
+from uuid import uuid4
+
+import pytest
+
+from portkey_ai import Portkey
+import inspect
+from tests.utils import read_json_file
+
+base_url = os.environ.get("PORTKEY_BASE_URL")
+api_key = os.environ.get("PORTKEY_API_KEY")
+virtual_api_key = os.environ.get("OPENAI_VIRTUAL_KEY")
+
+CONFIGS_PATH = "./tests/configs/threads"
+
+
+def get_configs(folder_path) -> List[Dict[str, Any]]:
+    config_files = []
+    for dirpath, _, file_names in os.walk(folder_path):
+        for f in file_names:
+            config_files.append(read_json_file(os.path.join(dirpath, f)))
+
+    return config_files
+
+
+class TestThreads:
+    client = Portkey
+    parametrize = pytest.mark.parametrize("client", [client], ids=["strict"])
+    models = read_json_file("./tests/models.json")
+
+    def get_metadata(self):
+        return {
+            "case": "testing",
+            "function": inspect.currentframe().f_back.f_code.co_name,
+            "random_id": str(uuid4()),
+        }
+
+    # # --------------------------
+    # # Test-1
+
+    # t1_params = []
```

Comment:
Please remove these comments if they are not necessary

Reply:
Done